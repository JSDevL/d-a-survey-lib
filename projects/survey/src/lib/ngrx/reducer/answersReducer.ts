import {Answer} from '../../model/answer/Answer';
import {AnswerAction} from '../action/answer/AnswerAction';

const defaultState = [];

export const answersReducer = (state: Answer[] = defaultState, action: AnswerAction) => {
  return (action && (action instanceof AnswerAction) && action.apply(state)) || state || defaultState;
};
