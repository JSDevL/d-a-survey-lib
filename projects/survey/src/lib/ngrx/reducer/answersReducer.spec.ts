import {answersReducer} from './answersReducer';

describe('answersReducer', () => {
  it('should return default state initially', () => {
    expect(answersReducer(null, null)).toMatchSnapshot();
  });

  it('should return state for unqualified action', () => {
    expect(answersReducer([null, null], null)).toEqual([null, null]);
  });
});
