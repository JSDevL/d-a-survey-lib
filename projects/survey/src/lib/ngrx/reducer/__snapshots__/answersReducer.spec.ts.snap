// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`answersReducer should return default state initially 1`] = `Array []`;
